src/cli/cli.cpp

```diff
@@ -3466,8 +3466,12 @@ otError Interpreter::ProcessMacFilterRss(int argc, char *argv[])
 #if OPENTHREAD_ENABLE_DIAG
 void Interpreter::ProcessDiag(int argc, char *argv[])
 {
+    char output[OPENTHREAD_CONFIG_DIAG_OUTPUT_BUFFER_SIZE];
+
     // all diagnostics related features are processed within diagnostics module
-    mServer->OutputFormat("%s\r\n", otDiagProcessCmd(argc, argv));
+    output[OPENTHREAD_CONFIG_DIAG_OUTPUT_BUFFER_SIZE - 1] = '\0';
```

Comment:
nit: how about using `sizeof(output)` instead of `OPENTHREAD_CONFIG_DIAG_OUTPUT_BUFFER_SIZE`?

Reply:
Updated, thanks.